js/adapt-hint.js

```diff
@@ -1,57 +1,19 @@
-define([
-  'core/js/adapt'
-], function (Adapt) {
+import Adapt from 'core/js/adapt';
+import HintView from './HintView';
 
-  const HintView = Backbone.View.extend({
+class Hint extends Backbone.Controller {
+  initialize() {
+    this.listenTo(Adapt, 'componentView:postRender', this.initHint);
+  }
 
-    events: {
-      'click .js-hint-btn-popup': 'onHintPopupClicked'
-    },
-
-    className: 'hint',
-
-    initialize: function() {
-      this.listenTo(Adapt, 'remove', this.remove);
-      this.render();
-    },
-
-    render: function() {
-      const data = this.model.toJSON();
-      const template = Handlebars.templates.hint;
-
-      const displayTitle = this.model.get('displayTitle');
-      const body = this.model.get('body');
-      const instruction = this.model.get('instruction');
-
-      if (displayTitle || body || instruction) {
-        this.$el.html(template(data)).appendTo($('.' + this.model.get('_id')).find('.component__header-inner'));
-      } else {
-        this.$el.html(template(data)).appendTo($('.' + this.model.get('_id')).find('.component__widget'));
-      }
-
-      _.defer(this.postRender.bind(this));
-    },
-
-    postRender: function() {
-      this.$el.parents('.component').addClass('has-hint');
-    },
-
-    onHintPopupClicked: function() {
-      Adapt.notify.popup({
-        ...this.model.get('_hint'),
-        _classes: 'hint-popup'
-      });
-    }
-
-  });
-
-  Adapt.on('componentView:postRender', function (view) {
+  initHint(view) {
```

Comment:
```suggestion
  onComponentPostRender(view) {
```
Name tells us about the lifecycle without having to refer to the midpoint of line 6